Fix counters in validator monitor totals mode

arnetheduck opened the pull request:
The current counters set gauges etc to the value of the _last_ validator
to be processed - as the name of the feature implies, we should be using
sums instead.

* fix missing beacon state metrics on startup, pre-first-head-selection

zah commented on beacon_chain/validators/validator_monitor.nim:
```diff
@@ -435,19 +474,43 @@ proc registerEpochInfo*(
+              epoch = current_epoch,
+              validator = id
+
+        if state.next_sync_committee.pubkeys.data.contains(pubkey):
```
How often is this code executed? There are ways to avoid the repeated per-validator linear scans here.

arnetheduck replied:
once per epoch, which is why I didn't bother for now - might do a followup but I want the metrics to be correct first, and this PR is messy enough

zah commented:
Needs rebasing

arnetheduck committed:
Fix counters in validator monitor totals mode

The current counters set gauges etc to the value of the _last_ validator
to be processed - as the name of the feature implies, we should be using
sums instead.

* fix missing beacon state metrics on startup, pre-first-head-selection
* fix epoch metrics not being updated on cross-epoch reorg
